blob-polyfill.ts

```diff
@@ -0,0 +1,99 @@
+import { getArrayBufferForBlob } from "react-native-blob-jsi-helper";
+import { ReadableStream } from "web-streams-polyfill";
+
+const decoder = new TextDecoder();
+
+/**
+ * React Native's Blob implementation has a constructor that currently only supports
+ * constructing from parts that are of type: Array<Blob | String>.
+ *
+ * Web5 packages need to be able to construct a Blob where one of the parts is of type `Uint8Array`.
+ *
+ * This function updates the constructor of Blob to convert any parts that are of type `Uint8Array`
+ * into a `string`, so that it can properly work with React Native's Blob implementation.
+ */
+function monkeyPatchBlobConstructor() {
+  const OriginalBlob = global.Blob;
+
+  const blobProxyHandler = {
+    construct(target: any, argumentsList: any) {
+      const blobParts = argumentsList[0];
+      const options = argumentsList[1];
+
+      if (blobParts) {
+        for (const [index, element] of blobParts.entries()) {
```

Comment:
how many times are we potentially looping here? should we do an upstream PR to fix this inside of react native itself?

Reply:
this is, potentially, very expensive code. if we are using blobs elsewhere we risk torpedoing the whole app's performance with this. i wonder what an upstream solution would involve?

Reply:
1. It'll loop as for as many entries that are in `blobParts`

2. I don't really know what an upstream solution would involve here. This was the majority of what took me a week to get this work done. Just figuring out that it was the problem was most of the battle. It's expensive, for sure. But it's working, and it can be our baseline for the time being.

Reply:
I think we should probably slap a TODO here and we should create an issue that tracks the TODO. We'll want visibility into the performance impact and it would be a good issue that could be contributed upstream by either OSS stakeholders or our own teams

Reply:
Added a task for us: https://github.com/TBD54566975/web5-wallet/issues/64
Referred to that task in a TODO in code here: [pr feedback](https://github.com/TBD54566975/web5-wallet/pull/62/commits/f6d6587fe7693ee2c3d46f47eccecef78a82ac1b)